File: Leetcode (Python, JavaScript)/3217. Delete Nodes From Linked List Present in Array/3217. Delete Nodes From Linked List Present in Array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        sets = set(nums)
        dummy = ListNode(0)
        dummy.next = head
        current = dummy
        while current.next:
            if current.next.val in sets:
                current.next = current.next.next
            else:
                current = current.next
        return dummy.next

File: Leetcode (Python, JavaScript)/3217. Delete Nodes From Linked List Present in Array/3217. Delete Nodes From Linked List Present in Array.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    const sets = new Set(nums);
    const dummy = new ListNode(0,head);
    let current = dummy;
    while (current.next!=null){
        if (sets.has(current.next.val)){
            current.next = current.next.next;
        }
        else{
            current = current.next;
        }
    }
    return dummy.next;
};